internal/dag/cache.go

```diff
@@ -84,24 +84,89 @@ func (kc *KubernetesCache) init() {
 	kc.extensions = make(map[types.NamespacedName]*contour_api_v1alpha1.ExtensionService)
 }
 
-// matchesIngressClass returns true if the given Kubernetes object
-// belongs to the Ingress class that this cache is using.
-func (kc *KubernetesCache) matchesIngressClass(obj metav1.Object) bool {
+// matchesIngressClass returns true if the given IngressClass
+// is the one this cache is using.
+func (kc *KubernetesCache) matchesIngressClass(obj *networking_v1.IngressClass) bool {
+	// If no ingress class name set, we allow an ingress class that is named
+	// with the default Contour accepted name.
+	if kc.IngressClassName == "" && obj.Name == "contour" {
```

Comment:
I wanted to pull the default ingress class name out of the `annotation` package but wasn't quite sure where to put it tbh

Reply:
I'm gonna have to pull some of the `IngressClass` matching logic out into something most likely, since I realized the `Ingress` status needs the same logic around status address updates